app/assets/javascripts/discourse/models/topic.js.es6

```diff
@@ -58,7 +59,13 @@ const Topic = RestModel.extend({
 
   @computed('fancy_title')
   fancyTitle(title) {
-    return censor(emojiUnescape(title || ""), Discourse.Site.currentProp('censored_words'));
+    let fancyTitle = censor(emojiUnescape(title || ""), Discourse.Site.currentProp('censored_words'));
+
+    if (Discourse.SiteSettings.support_mixed_text_direction) {
```

Comment:
This should be `this.siteSettings` - although the `Discourse.Site` above makes me wonder if it's not wired up properly. If it doesn't work falling back to `Discourse.SiteSettings` is fine.